src/Aquifer.API/Endpoints/Bibles/Versification/Endpoint.cs

```diff
@@ -0,0 +1,192 @@
+        
+        var isRequestedStartVerseValid = false;
+        var isRequestedEndVerseValid = false;
+
+        if (req.StartVerse.HasValue)
+        {
+            isRequestedStartVerseValid = await IsVerseValidForBookAndChapterAsync(req.StartVerse.Value, minChapterNumber, req.BookId, ct);
+        }
+
+        if (req.EndVerse.HasValue)
+        {
+            isRequestedEndVerseValid = await IsVerseValidForBookAndChapterAsync(req.EndVerse.Value, minChapterNumber, req.BookId, ct);
+        }
+        
+        if (isRequestedStartVerseValid && isRequestedEndVerseValid)
+        {
+            return (BibleUtilities.GetVerseId((BookId)req.BookId, minChapterNumber, req.StartVerse!.Value),
+                    BibleUtilities.GetVerseId((BookId)req.BookId, maxChapterNumber, req.EndVerse!.Value));
+        }
+        
+        var maxVerse = await dbContext.BookChapters
+            .Where(bc => bc.BookId == (BookId)req.BookId && bc.Number == maxChapterNumber)
+            .Select(bc => bc.MaxVerseNumber)
+            .FirstOrDefaultAsync(ct);
+        
+        if (isRequestedStartVerseValid && !isRequestedEndVerseValid)
+        {
+            return (BibleUtilities.GetVerseId((BookId)req.BookId, minChapterNumber, req.StartVerse!.Value),
+                    BibleUtilities.GetVerseId((BookId)req.BookId, maxChapterNumber, maxVerse));
+        }
+        
+        var minVerse = await dbContext.BibleTexts
+            .Where(bt => bt.BookId == (BookId)req.BookId && bt.ChapterNumber == minChapterNumber)
+            .OrderBy(bt => bt.VerseNumber)
+            .Select(bt => bt.VerseNumber)
+            .FirstOrDefaultAsync(ct);
+
+        if (!isRequestedStartVerseValid && isRequestedEndVerseValid)
+        {
+            return (BibleUtilities.GetVerseId((BookId)req.BookId, minChapterNumber, minVerse),
+                    BibleUtilities.GetVerseId((BookId)req.BookId, maxChapterNumber, req.EndVerse!.Value));
+        }
+        
+        return (BibleUtilities.GetVerseId((BookId)req.BookId, minChapterNumber, minVerse),
+                BibleUtilities.GetVerseId((BookId)req.BookId, maxChapterNumber, maxVerse));
+    }
+    
+    private static VerseReference FormatBaseBookChapterVerseMapping(int mappedVerseId)
+    {
+        var (targetBookId, targetChapter, targetVerse) = BibleUtilities.TranslateVerseId(mappedVerseId);
+        var targetBookName = BibleBookCodeUtilities.FullNameFromId(targetBookId);
+        
+        return new VerseReference
+        {
+            VerseId = mappedVerseId,
+            Book = targetBookName,
+            Chapter = targetChapter,
+            Verse = targetVerse
+        };
+    }
+
+    private static bool IsRequestedBookIdValid(int bookId)
+    {
+        return Enum.IsDefined(typeof(BookId), bookId);
+    }
+
+    private async Task<bool> IsChapterValidForBookAsync(int chapterNumber, int bookId, CancellationToken ct)
+    {
+        return await dbContext.BookChapters
```

Comment:
FYI, `BookChapters` only corresponds to Bible ID `0` (`eng` versification).  This is correct for now but if we ever want to pass source Bible ID then this will be wrong.  It would be better to check `BibleTexts`.

Reply:
Hold off on changing things, though.  I have an idea and am writing up another comment.